plugin/src/main/java/com/redhat/jenkins/nodesharingfrontend/SharedOnceRetentionStrategy.java

```diff
@@ -83,6 +85,13 @@ private void done(Executor executor) {
     @SuppressFBWarnings(value="SE_BAD_FIELD", justification="not a real Callable")
     private void done(final AbstractCloudComputer<?> c) {
         c.setAcceptingTasks(false); // just in case
+        if (c.isOffline() && c.getOfflineCause() instanceof OfflineCause.UserCause) {
```

Comment:
Note this cover computer "disconnected" by user, not only those that are "temporarily offline". Not sure this is a problem.

Reply:
Logic and common sense tell me it is OK, user explicitly wants to disable automatic release of the node after job execution, it doesn't matter either by temporary offline or disconnecting the machine (as we do know, disconnecting a machine doesn't implies loosing the background channel). I think, it'd be better to cover both 'offline' scenarios in the same manner.

Reply:
Agreed